runtimes/ocaml/runtime.ml

```diff
@@ -703,61 +722,60 @@ end
 
 let handle_default :
       'a.
-      source_position ->
+      source_position array ->
       (unit -> 'a) array ->
       (unit -> bool) ->
       (unit -> 'a) ->
       'a =
  fun pos exceptions just cons ->
-  let except =
-    Array.fold_left
-      (fun acc except ->
-        let new_val = try Some (except ()) with EmptyError -> None in
-        match acc, new_val with
-        | None, _ -> new_val
-        | Some _, None -> acc
-        | Some _, Some _ -> raise (ConflictError pos))
-      None exceptions
+  let len = Array.length exceptions in
+  let rec filt_except i =
+    if i < len then
+      match exceptions.(i) () with
+      | new_val -> (new_val, i) :: filt_except (i + 1)
+      | exception Empty -> filt_except (i + 1)
+    else []
```

Comment:
Did you rewrite into a tail-call form to get more performance?

Reply:
Hm, the main reason for rewriting was to be able to report conflicts between more than 2 exceptions (here again, to match the reporting done in the interpreter).

The new version is actually not tail-recursive, while the original was (using `Array.fold`). Why ?
- There is no `Array.fold2` which would have been the simplest choice
- We have no recursion in Catala, so there is no risk that the number of exceptions gets anywhere near the stack limit
- The non-tail recursive version is more efficient

In addition, if we're still worried about stack size, since we already require OCaml 4.14, we could annotate with `[@ocaml.tail_mod_cons]` and get both bounded stack space and efficiency.